src/Algolia.Search/Serializer/IndexSettingsConverter.cs

```diff
@@ -0,0 +1,93 @@
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Algolia.Search.Models.Settings;
+
+namespace Algolia.Search.Serializer
+{
+    /// <summary>
+    /// Algolia's custom converter for IndexSettings
+    /// This could be deleted when System.Text.Json while handle field
+    /// This converter also has it's own Serializer option. You might modify this one as well
+    /// </summary>
+    public class IndexSettingsConverter : JsonConverter<IndexSettings>
+    {
+        /// <summary>
+        /// The converter requires it's own options instance to avoid circular calls.
+        /// </summary>
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions
+        {
+            IgnoreNullValues = true,
+            WriteIndented = false,
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            Converters =
+            {
+                new ObjectToInferredTypesConverter(), new NullableDateTimeConverter(), new DateTimeConverter()
+            }
+        };
+
+        /// <summary>
+        /// Custom reader for IndexSettings. Handle legacy settings
+        /// </summary>
+        public override IndexSettings Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var settings =
+                JsonSerializer.Deserialize<IndexSettings>(ref reader, _serializerOptions);
+
+            if (settings.CustomSettings.ContainsKey("attributesToIndex") &&
+                settings.CustomSettings["attributesToIndex"] != null)
+            {
+                settings.SearchableAttributes = (List<string>)settings.CustomSettings["attributesToIndex"];
+                settings.CustomSettings.Remove("attributesToIndex");
+            }
+
+            if (settings.CustomSettings.ContainsKey("slaves") && settings.CustomSettings["slaves"] != null)
+            {
+                settings.Replicas = (List<string>)settings.CustomSettings["slaves"];
+                settings.CustomSettings.Remove("slaves");
+            }
+
+            if (settings.CustomSettings.ContainsKey("numericAttributesToIndex") &&
+                settings.CustomSettings["numericAttributesToIndex"] != null)
+            {
+                settings.NumericAttributesForFiltering =
+                    (List<string>)settings.CustomSettings["numericAttributesToIndex"];
+                settings.CustomSettings.Remove("numericAttributesToIndex");
+            }
+
+            return settings;
+        }
+
+        /// <summary>
+        /// Using default writer there.
+        /// </summary>
+        public override void Write(Utf8JsonWriter writer, IndexSettings value, JsonSerializerOptions options)
```

Comment:
Same question as before, can't we get rid of the `Write()` method override as we want to have the default serializer?